api/types/constants.go

```diff
@@ -425,6 +425,10 @@ const (
 
 	// BotGenerationLabel is a label used to record the certificate generation counter.
 	BotGenerationLabel = "teleport.internal/bot-generation"
+
+	// InternalResourceIDLabel is a label used to store an ID to correlate between two resources
+	// A pratical example of this is to create a correlation between a Node Provision Token and the Node that used that token to join the cluster
+	InternalResourceIDLabel = "teleport.internal/resource-id"
```

Comment:
Should this be `teleport.dev/resource-id` instead? Or are we deprecating `teleport.dev` and standardizing on `teleport.internal` instead?

Reply:
I think the idea is to keep both and filter out `teleport.internal` labels from the UI (not sure if that's the case now).

Reply:
So, the WebUI doesn't filter `teleport.dev` as well
![image](https://user-images.githubusercontent.com/689271/182809858-6d833458-af01-4d97-9055-27a5055433f7.png)

I'm not sure which one we should use, but after looking into this code
https://github.com/gravitational/teleport/blob/54616acce4f05876aa22077b677c6460483e3568/tool/tctl/common/collection.go#L804

I would say we should use `teleport.dev/` and create an issue to filter out these labels when displaying labels in the UI

Reply:
I'd say `teleport.internal/` should be hidden, `teleport.dev/` is also used for stuff like automatic labels from LDAP for desktop access - so I'm ok with `teleport.internal/resource-id`.

What I'm not ok with is that one bot label using `teleport.internal/kebab-case` while desktop access uses `teleport.dev/snake_case`, but that's not something that can be fixed here (or ever? 😭)

Reply:
I'll keep it as `teleport.internal/` then.

As for case, it seems `teleport.internal/` uses kebab case and `teleport.dev/` uses snake case
At least they are consistent within the namespace :joy: 
Although the `teleport.internal/` only has one occurrence, so we can't really say if that's the rule or the exception :sweat_smile: